### PR TITLE
Apply design system & edit interface codes

### PR DESCRIPTION
**Pull request by soyeongmin**
1. Apply design system(typo, color, basic components) in the codes
2. Edit responsive UI (mobile phone ver.)

### CODE REVIEW

File: src/Components/basic/button/TagButton.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import styled from '@emotion/styled'
+import { css } from '@emotion/react'
+import { theme } from '../../../styles/theme'
+
+interface tagProps {
+  onClick: () => void
+  type: string
+  id: string
```
Comment (jhw123) [inline src/Components/basic/button/TagButton.tsx:8]:
```suggestion
  id: "AnsAct" | "Ans" | "Dist" | "DistAct"
```

File: src/Components/TextEditor.tsx [hunk 2]
```diff
@@ -1,43 +1,38 @@
...
+        inline: { inDropdown: true },
+        list: { inDropdown: true },
+        textAlign: { inDropdown: true },
+        link: { inDropdown: true },
+      }}
+    />
+  )
+}
 
 const editorStyle = {
```
Comment (jhw123) [inline src/Components/TextEditor.tsx:33]: You can also do
```suggestion
const editorStyle = styled(Editor)`
border: `1px solid ${theme.palette.grey[500]}`,
   padding: 16px;
   border-radius: 6px;
   height: 200px;
`
```

File: src/Components/CreateNewOption.tsx [hunk 3]
```diff
@@ -12,87 +15,31 @@ export const CreateNewOption = () => {
...
-      </Toggles>
-      <div>
-        <Input type="text" placeholder="Suggest an answer or distractor for this question" />
-        <CategoryInput getCategory={getCategory} />
-        <SubmitBtn disabled>Submit</SubmitBtn>
-      </div>
+      <Container>
+        <div>
+          <Label text="Create New Option" color="blue" size={0} />
+          <TagBtn onClick={() => setTag('Ans')} id={tag == 'Ans' ? 'AnsAct' : 'Ans'} type="Answer" />
```
Comment (jhw123) [inline src/Components/CreateNewOption.tsx:21]: if tag has two possible values only, it is better to use boolean.
```typescript
const [isAns, setIsAns] = useState(true)
```

File: src/Components/basic/button/OptionButton.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+const OptionBtnComponent = styled.div<{ state: boolean; selected: boolean }>`
+  ${({ state, selected }) => css`
+    ${theme.typography.b02};
+    background-color: ${theme.palette.background.light};
+    padding: 16px;
+    margin-bottom: 8px;
+    border-radius: 6px;
+    border: 1.5px solid rgba(0, 0, 0, 0);
```
Comment (jhw123) [inline src/Components/basic/button/OptionButton.tsx:27]: you can also define border color in your design system
```suggestion
    border: 1.5px solid ${theme.palette.border.black};
```

File: .eslintrc.js [hunk 5]
```diff
@@ -12,6 +12,7 @@ module.exports = {
   },
   plugins: ['react', '@typescript-eslint'],
   rules: {
+    'react/no-unknown-property': ['error', { ignore: ['css'] }],
```
Comment (jhw123) [inline .eslintrc.js:15]: why did have to you turn this off?

File: package.json [hunk 6]
```diff
@@ -38,6 +38,7 @@
     "redux-logger": "^3.0.6",
     "redux-persist": "^6.0.0",
     "redux-thunk": "^2.4.2",
+    "type-check": "^0.4.0",
```
Comment (jhw123) [inline package.json:41]: where do you use this library?

File: src/Components/basic/button/Button.tsx [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+interface buttonProps {
+  text: string
+  onClick?: () => void
+  disabled?: boolean
+}
+
+export const FillBtn = (props: buttonProps) => {
+  return (
+    <Fill onClick={props.onClick} disabled={props.disabled}>
+      {props.text}
```
[the remark this answers is not in the document]
Comment (soyeongmin) [reply src/Components/basic/button/Button.tsx:13]: Hmm.. I'm not sure what you means. Can you explain it more?

File: src/styles/theme.tsx [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+  overline: css`
+    font-family: 'inter-r';
+    font-size: 12px;
+    @media (max-width: 599px) {
+      font-size: 11px;
+    }
+  `,
+}
+
+export const theme = {
```
[the remark this answers is not in the document]
Comment (soyeongmin) [reply src/styles/theme.tsx:111]: It is okay to separate the theme for each system, but there are not many design system variations. Do you think dividing them is better?

File: src/styles/globalStyle.tsx [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    src: url(${interM});
+  }
+  @font-face {
+    font-family: 'inter-sb';
+    src: url(${interSb});
+  }
+  html {
+    background-color: #e6eaef;
+  }
+  .App {
```
[the remark this answers is not in the document]
Comment (soyeongmin) [reply src/styles/globalStyle.tsx:27]: okay!

File: src/Components/basic/button/Button.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (jhw123) [reply src/Components/basic/button/Button.tsx:13]: oh I meant you could design the interface like
```typescript
interface buttonProps {
  children: React.ReactNode
}
```
and do
```tsx
export const FillBtn = (props: buttonProps) => {
  return (
    <Fill onClick={props.onClick} disabled={props.disabled}>
      {props.children}
    </Fill>
  )
}
```
This makes the component more usable and flexible as you can pass text or other components
```tsx
<FillBtn>this is a button</FillBtn>

<FillBtn><img src="..."/>this is a button</FillBtn>
```

File: src/index.tsx [hunk 10]
```diff
@@ -7,17 +7,19 @@ import { Provider } from 'react-redux'
 import { persistStore } from 'redux-persist'
 import { PersistGate } from 'redux-persist/integration/react'
 import { store } from './state/store'
+import { render } from 'react-dom'
 
 const root = ReactDOM.createRoot(document.getElementById('root') as HTMLElement)
 const persistor = persistStore(store)
 
-root.render(
```
[the remark this answers is not in the document]
Comment (soyeongmin) [reply src/index.tsx:14]: It was for Text Editor. I added react-draft-wysiwyg libraries for the text editor, but if the previous codes in index.tsx, the dropdown function is not working. After searching for the reasons for that, fixing codes like this is the only working solution.

File: src/Components/Gnb.tsx [hunk 11]
```diff
@@ -14,13 +19,41 @@ export const Gnb = (props: { loginState: boolean }) => {
...
       {props.loginState == true && (
         <>
-          <Menu>
-            <MenuBtn onClick={() => navigate('/solve')}>Solve Problem</MenuBtn>
-            <MenuBtn onClick={() => navigate('/createQuestion')}>Create Question</MenuBtn>
-            <MenuBtn onClick={() => navigate('/question/createOption')}>Create Options</MenuBtn>
-            <MenuBtn onClick={() => navigate('/mypage')}>My Page</MenuBtn>
+          <Menu isDisplay={isDisplay}>
+            <MenuBtn
+              onClick={() => {
```
[the remark this answers is not in the document]
Comment (soyeongmin) [reply src/Components/Gnb.tsx:24]: Oh this will be much nicer. Yep!

File: src/Components/basic/button/TagButton.tsx (hunk 1 above)
Comment (soyeongmin) [reply src/Components/basic/button/TagButton.tsx:8]: Aha... I didn't think like this. Thanks

File: src/Components/TextEditor.tsx (hunk 2 above)
Comment (soyeongmin) [reply src/Components/TextEditor.tsx:33]: Oh I didn't know it. I will try :)

File: src/Components/CreateNewOption.tsx (hunk 3 above)
Comment (soyeongmin) [reply src/Components/CreateNewOption.tsx:21]: Maybe the tag will be 4 names, so I will check and if it is possible, then I will replace it.

File: src/Components/basic/button/OptionButton.tsx (hunk 4 above)
Comment (soyeongmin) [reply src/Components/basic/button/OptionButton.tsx:27]: Actually, it is for transparent color not to change the size of components when changing the state(before & after clicking). THen, I need to add the color.

File: .eslintrc.js (hunk 5 above)
Comment (soyeongmin) [reply .eslintrc.js:15]: I tried to fix the error in 'CSS' props in HTML tags like
`<div **css**={css``}>/div>`
THe bolding CSS makes not known property error. After ignoring codes like this, the error disappeared but still have a problem, so I didn't use the codes. I will delete it.

File: package.json (hunk 6 above)
Comment (soyeongmin) [reply package.json:41]: Hmm... maybe I installed it for fixing 'CSS' property from StackOverflow. It didn't work.....:(

File: src/Components/basic/button/Button.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (soyeongmin) [reply src/Components/basic/button/Button.tsx:13]: Oh I didn't know the way. Actually, I thought passing all the props is not a good method but didn't know how to do it efficiently. I will fix it! Thanks ;)

File: src/styles/theme.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (jhw123) [reply src/styles/theme.tsx:112]: I think yes? if we separate them, we can type less
```tsx
${theme.typography.logo}
// to
${typography.logo}
```

[the remark this answers is not in the document]
Comment (soyeongmin) [reply src/styles/theme.tsx:112]: okay!

**Pull request merged**